### ISSUE TITLE
Pilot manager cannot expand a rcfg entry

### ISSUE THREAD

**Issue by AymenFJA** [radical-cybertools/radical.pilot#1978]
```
  python               : 3.7.0
  pythonpath           : 
  virtualenv           : /home/aymen/rp_test

  radical.pilot        : 0.90.1
  radical.saga         : 0.90.0
  radical.utils        : 0.90.3
```

```
Traceback (most recent call last):
  File "/home/aymen/rp2_test/lib/python3.7/site-packages/radical/pilot/pmgr/launching/default.py", line 486, in work
    self._start_pilot_bulk(resource, schema, pilots)
  File "/home/aymen/rp2_test/lib/python3.7/site-packages/radical/pilot/pmgr/launching/default.py", line 566, in _start_pilot_bulk
    rcfg[k]  = rcfg[k] % expand
ValueError: unsupported format character ''' (0x27) at index 47
```

**Comment by iparask** [comment]
This is also happening with the python 2 stack:
```

  python               : 2.7.15
  pythonpath           :
  virtualenv           : rp_spark

  radical.pilot        : 0.73.1-v0.73.1-3-ga8acaad5@python2-devel
  radical.saga         : 0.72.1-v0.72.1@python2-devel
  radical.utils        : 0.72.0-v0.72.0@python2-devel
```

Should I try to take a crack at it or @AymenFJA you are looking at it?

**Comment by AymenFJA** [comment]
Based on the `last devel call` with the RCT-Devel team @andre-merzky suggested putting log probs : 

The following is the output :
```
2019-10-25 11:25:57,633: pmgr.0000.launching.0: pmgr.0000.launching.0           : MainThread     : DEBUG   : RCFG key: agent_config, rcfg: default
2019-10-25 11:25:57,633: pmgr.0000.launching.0: pmgr.0000.launching.0           : MainThread     : DEBUG   : RCFG key: agent_launch_method, rcfg: ORTE
2019-10-25 11:25:57,633: pmgr.0000.launching.0: pmgr.0000.launching.0           : MainThread     : DEBUG   : RCFG key: agent_scheduler, rcfg: CONTINUOUS
2019-10-25 11:25:57,633: pmgr.0000.launching.0: pmgr.0000.launching.0           : MainThread     : DEBUG   : RCFG key: agent_spawner, rcfg: POPEN
2019-10-25 11:25:57,633: pmgr.0000.launching.0: pmgr.0000.launching.0           : MainThread     : DEBUG   : RCFG key: cores_per_node, rcfg: 24
2019-10-25 11:25:57,633: pmgr.0000.launching.0: pmgr.0000.launching.0           : MainThread     : DEBUG   : RCFG key: default_queue, rcfg: compute
2019-10-25 11:25:57,634: pmgr.0000.launching.0: pmgr.0000.launching.0           : MainThread     : DEBUG   : RCFG key: description, rcfg: The Comet HPC resource at SDSC 'HPC for the 99%' (http://www.sdsc.edu/services/hpc/hpc_systems.html#comet).
2019-10-25 11:25:57,634: pmgr.0000.launching.0: pmgr.0000.launching.0           : MainThread     : ERROR   : bulk launch failed
Traceback (most recent call last):
  File "/home/aymen/rp2_test/lib/python3.7/site-packages/radical/pilot/pmgr/launching/default.py", line 484, in work
    self._start_pilot_bulk(resource, schema, pilots)
  File "/home/aymen/rp2_test/lib/python3.7/site-packages/radical/pilot/pmgr/launching/default.py", line 566, in _start_pilot_bulk
    rcfg[k]  = rcfg[k] % expand
ValueError: unsupported format character ''' (0x27) at index 47
```

It appears that the problem is in this line ` RCFG key: description, rcfg: The Comet HPC resource at SDSC 'HPC for the 99%' (http://www.sdsc.edu/services/hpc/hpc_systems.html#comet).`  

if we check the error above it's showing only  3 single quotes ` '''  ` not 2 (1 string)
and not 4 (2 strings).

**Comment by andre-merzky** [comment]
Hmm, I don't yet see how that breaks:
```python
>>> d={'foo': 'bar'}
>>> s = "%(foo)s ''' bar"
>>> s % d
"bar ''' bar"
```
Any idea how to reduce this to a reproducible case?  Thanks!

**Comment by andre-merzky** [comment]
got it:
```python
>>> s = "%s '99%' buz"
>>> s % ('foo', 'baz')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: unsupported format character ''' (0x27) at index 7
>>> s = "%s '99%%' buz"
>>> s % ('foo')
"foo '99%' buz"
```
So the `%` character in the description needs to be escaped.   Can you please provide a PR?  Thanks!1

**Comment by AymenFJA** [comment]
@andre-merzky  sorry for the late response. Sure, I will do that after today's devel call.

**Comment by AymenFJA** [comment]
This ticket is solved with  this PR [fix/#1978](https://github.com/radical-cybertools/radical.pilot/pull/1992)
